pennylane/devices/experimental/device_api.py

```diff
@@ -0,0 +1,555 @@
+
+        Utility functions will be added to the ``devices`` module to query whether or not the device driver can do certain things, such
+        as ``devices.supports_operator(op, dev, native=True)``. These functions will work by checking the behaviour of :meth:`~.Device.preprocess`
+        to certain inputs.
+
+        Versioning should be specified by the package containing the device. If an external package includes a PennyLane device,
+        then the package requirements should specify the minimium PennyLane version required to work with the device.
+
+
+    .. details::
+        :title: Execution Configuration
+
+        Execution config properties related to configuring a device include:
+
+        * ``device_options``: A dictionary of device specific options. For example, the python device may have ``multiprocessing_mode``
+          as a key. These should be documented in the class docstring.
+
+        * ``gradient_method``: A device can choose to have native support for any type of gradient method. If the method
+          :meth:`~.supports_derivatives` returns ``True`` for a particular gradient method, it will be treated as a device
+          derivative and not handled by pennylane core code.
+
+        * ``gradient_keyword_arguments``: Options for the gradient method.
+
+        * ``derivative_order``: Relevant for requested device derivatives.
+
+    """
+
+    @property
+    def name(self) -> str:
+        """The name of the device or set of devices.
+
+        This property can either be the name of the class, or an alias to be used in the :func:`~.device` constructor,
+        such as ``"default.qubit"`` or ``"lightning.qubit"``.
+
+        """
+        return type(self).__name__
+
+    tracker: Tracker = Tracker()
+    """A :class:`~.Tracker` that can store information about device executions, shots, batches,
+    intermediate results, or any additional device dependent information.
+
+    A plugin developer can store information in the tracker by:
+
+    .. code-block:: python
+
+        # querying if the tracker is active
+        if self.tracker.active:
+
+            # store any keyword: value pairs of information
+            self.tracker.update(executions=1, shots=self._shots, results=results)
+
+            # Calling a user-provided callback function
+            self.tracker.record()
+    """
+
+    def __init__(self, *args, **kwargs) -> None:
+        # each instance should have its own Tracker.
+        self.tracker = Tracker()
+
+    def preprocess(
+        self,
+        circuits: QuantumTape_or_Batch,
+        execution_config: ExecutionConfig = DefaultExecutionConfig,
+    ) -> Tuple[QuantumTapeBatch, Callable]:
+        """Device preprocessing function.
+
+        .. warning::
+
+            This function is tracked by machine learning interfaces and should be fully differentiable.
+            The ``pennylane.math`` module can be used to construct fully differntiable transformations.
```

Comment:
In previous conversations it sounded like you did not want to make pennylane.math part of the "public api" and support it. I take from this that this has changed and that pennylane.math will be a considered part of the official pennylane user interface?

Reply:
Hi @cvjjm! This is a good question. Right now we are thinking that `pennylane.math` is not intended to be generally user facing. The exception to this, as you highlight, may be for _developer_ users who build plugins - who will need to override methods such as `preprocess`  which may need use of `pennylane.math` or `autoray` to maintain differentiability with all interfaces.

So in short, `pennylane.math` will not be "advertised" to most users and may not be completely stable, but it's also a fairly important part of PennyLane's internals for now.